consensus/qbft-core/src/main/java/org/hyperledger/besu/consensus/qbft/core/statemachine/QbftBlockHeightManager.java

```diff
@@ -155,11 +155,11 @@ private void buildBlockAndMaybePropose(
     }
 
     final long headerTimeStampSeconds = Math.round(clock.millis() / 1000D);
-    final Block block = qbftRound.createBlock(headerTimeStampSeconds);
-    final boolean blockHasTransactions = !block.getBody().getTransactions().isEmpty();
-    if (blockHasTransactions) {
+    final QbftBlock block = qbftRound.createBlock(headerTimeStampSeconds);
+    final boolean blockIsEmpty = block.isEmpty();
+    if (!blockIsEmpty) {
```

Comment:
nit: `if (!block.isEmpty())`

Reply:
done, also change the code back to use the existing `block.getBody().getTransactions().isEmpty()` check we doing before to determine if the block is empty. will spin up a quick testnet and verify that the less empty block flag works, too after the change.